purchase_mto_owner/models/stock_rule.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2024 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockRule(models.Model):
+    _inherit = "stock.rule"
+
+    def _prepare_purchase_order(self, company_id, origins, values):
+        vals = super(StockRule, self)._prepare_purchase_order(
+            company_id, origins, values
+        )
+        values = values[0]
+        move_dest = values.get("move_dest_ids")
```

Comment:
@AungKoKoLin1997 Did you test this code? I can't find the relevant code where `move_dest_ids` is assigned in `values`. The following line throws a server error because there is no `move_dest`.

Reply:
> I can't find the relevant code where `move_dest_ids` is assigned in `values`.

https://github.com/OCA/OCB/blob/b1c66f1/addons/stock/models/stock_move.py#L1422

`move_dest_ids` is assigned in the MTO scenario, not in the scenario of orderpoint procurements, which is not considered in the current code.